exporter/stackdriverexporter/stackdriver.go

```diff
@@ -121,6 +122,7 @@ func newStackdriverMetricsExporter(cfg *Config) (component.MetricsExporter, erro
 		if err != nil {
 			return nil, err
 		}
+		options.TraceClientOptions = copts
```

Comment:
Do we need this for metric exporter?

Reply:
This is a bit weird but essentially it's because:

- In OT, the trace & metric packages are split
- In OC, the trace & metric packages are combined, but we now only use the metric part (trace has migrated to OT) ==> when we create an OC exporter it creates both a trace & metric exporter
- When creating a trace or metric client, if the Endpoint & UseInsecure options are not set, the client will try to look up ServiceAccount creds and fail (i.e. during tests) ==> when creating the OC exporter for testing purposes, we need to set both trace & metric client options